From 732bccb58abb383c954f4831bfa587f430788370 Mon Sep 17 00:00:00 2001
From: Myron Stowe <myron.stowe@redhat.com>
Date: Thu, 10 Jul 2014 00:58:47 -0400
Subject: [acpi] dock: simplify dock_create_acpi_device()

Message-id: <20140710005847.22146.76225.stgit@gir.stowe>
Patchwork-id: 85544
O-Subject: [RHEL7.1 PATCH 005/105] ACPI / dock: simplify dock_create_acpi_device()
Bugzilla: 1114228
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1114228
Upstream Status: 472d963befe28b8614ea2789757b27536c8d79eb

commit 472d963befe28b8614ea2789757b27536c8d79eb
Author: Jiang Liu <jiang.liu@huawei.com>
Date:   Sat Jun 29 00:24:37 2013 +0800

    ACPI / dock: simplify dock_create_acpi_device()

    The return value of dock_create_acpi_device() is not used at all,
    so change its signature to return void and simplify the
    implementation of it.

    Signed-off-by: Jiang Liu <jiang.liu@huawei.com>
    Signed-off-by: Rafael J. Wysocki <rafael.j.wysocki@intel.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/acpi/dock.c b/drivers/acpi/dock.c
index f3ec722..1bdb1fa 100644
--- a/drivers/acpi/dock.c
+++ b/drivers/acpi/dock.c
@@ -351,10 +351,8 @@ static int dock_present(struct dock_station *ds)
  *  handle if one does not exist already.  This should cause
  *  acpi to scan for drivers for the given devices, and call
  *  matching driver's add routine.
- *
- *  Returns a pointer to the acpi_device corresponding to the handle.
  */
-static struct acpi_device * dock_create_acpi_device(acpi_handle handle)
+static void dock_create_acpi_device(acpi_handle handle)
 {
  struct acpi_device *device;
  int ret;
@@ -367,10 +365,7 @@ static struct acpi_device * dock_create_acpi_device(acpi_handle handle)
   ret = acpi_bus_scan(handle);
   if (ret)
    pr_debug("error adding bus, %x\n", -ret);
-
-  acpi_bus_get_device(handle, &device);
  }
- return device;
 }
 
 /**
-- 
1.7.1